framework/filters/AccessControl.php

```diff
@@ -113,26 +118,37 @@ public function beforeAction($action)
     {
         $user = $this->user;
         $request = Yii::$app->getRequest();
+
         /* @var $rule AccessRule */
         foreach ($this->rules as $rule) {
-            if ($allow = $rule->allows($action, $user, $request)) {
+            $allow = $rule->allows($action, $user, $request);
+
+            if ($allow) {
                 return true;
             } elseif ($allow === false) {
```

Comment:
Could be just `if ($allow === false) {` since previously we're returning immediately.